### PR TITLE
FALCON-2233, Changes in falcon unit to enable/disable extensions

### PR DESCRIPTION
**Pull request by PracheerAgarwal**
1. Changes in falcon unit to enable/disable extensions
2. Adding enable/disable api in server
3. check for entension and its job existence
4. Few indentation changes

### CODE REVIEW

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 1]
```diff
@@ -586,6 +597,10 @@ public APIResult update(
         String extensionName = getExtensionName(jobName);
         try {
             entityMap = getEntityList(extensionName, jobName, feedForms, processForms, config);
+            if (entityMap.isEmpty()) {
+                // return failure if the extension job doesn't exist
```
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: entityMap won't be empty. But the list of values in them can be empty. One of them can be empty. Both may not be empty. Can have a check on the same for both not being empty.

File: prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java [hunk 2]
```diff
@@ -232,4 +232,23 @@ private static JSONArray buildEnumerateResult() throws FalconException {
...
+    public static void checkIfExtensionIsEnabled(String extensionName) {
+        ExtensionMetaStore metaStore = ExtensionStore.getMetaStore();
+        if (!metaStore.getDetail(extensionName).getStatus().equals(ExtensionStatus.ENABLED)) {
+            LOG.error("Extension: " + extensionName + " is in disabled state.");
+            throw FalconWebException.newAPIException("Extension: " + extensionName + " is in disabled state.",
+                    Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    public static void checkIfExtensionJobExists(String jobName, String extensionName) {
```
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java]: You can keep them protected.

File: unit/src/main/java/org/apache/falcon/unit/LocalExtensionManager.java [hunk 3]
```diff
@@ -40,26 +40,27 @@
...
-    LocalExtensionManager() {}
+    LocalExtensionManager() {
+    }
 
     APIResult submitExtensionJob(String extensionName, String jobName, InputStream configStream,
-                                 SortedMap<EntityType, List<Entity>> entityMap)
-        throws FalconException, IOException {
+                                 SortedMap<EntityType, List<Entity>> entityMap) throws FalconException, IOException {
+        checkIfExtensionIsEnabled(extensionName);
+        checkIfExtensionJobExists(jobName, extensionName);
```
Comment (sandeepSamudrala) [inline unit/src/main/java/org/apache/falcon/unit/LocalExtensionManager.java]: Job will exist post submit. You can have this check in schedule.

File: unit/src/test/java/org/apache/falcon/unit/TestFalconUnit.java [hunk 4]
```diff
@@ -442,8 +442,18 @@ public void testExtensionJobOperations() throws Exception {
         String result = registerExtension(TEST_EXTENSION, STORAGE_URL + EXTENSION_PATH, TEST_EXTENSION);
         Assert.assertEquals(result, "Extension :testExtension registered successfully.");
 
+        disableExtension(TEST_EXTENSION);
         createDir(PROCESS_APP_PATH);
         copyExtensionJar(packageBuildLib);
+
+        try {
+            submitExtensionJob(TEST_EXTENSION, TEST_JOB, null, null);
```
Comment (sandeepSamudrala) [inline unit/src/test/java/org/apache/falcon/unit/TestFalconUnit.java]: Assert.fail("should have thrown exception saying extension is disabled");

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java (hunk 1 above)
Comment (PracheerAgarwal) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: ack.

File: prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java (hunk 2 above)
Comment (PracheerAgarwal) [inline prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java]: changed.

File: unit/src/test/java/org/apache/falcon/unit/TestFalconUnit.java (hunk 4 above)
Comment (PracheerAgarwal) [inline unit/src/test/java/org/apache/falcon/unit/TestFalconUnit.java]: done.

**Pull request closed without merging**